File: output/llama3.1-8b_IO_1/avatar/Java/Python/codeforces_242_B.py
import sys

INF = 10**9 + 5
scanner = sys.stdin.readline

def main():
    n = int(scanner())
    a = []
    b = []

    for _ in range(n):
        line = scanner().split()
        a.append(int(line[0]))
        b.append(int(line[1]))

    left, right = INF, 0
    for i in range(n):
        left = min(left, a[i])
        right = max(right, b[i])

    for i in range(n):
        if left == a[i] and right == b[i]:
            print(i + 1)
            return

    print(-1)

if __name__ == "__main__":
    main()

File: output/llama3.1-8b_IO_1/avatar/Python/Java/codeforces_242_B.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class codeforces_242_B {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int id = 0;
        String[] input1 = br.readLine().split(" ");
        int l = Integer.parseInt(input1[0]);
        int r = Integer.parseInt(input1[1]);

        for (int i = 1; i <= n; i++) {
            String[] input2 = br.readLine().split(" ");
            int li = Integer.parseInt(input2[0]);
            int ri = Integer.parseInt(input2[1]);
            if (li <= l && r <= ri) id = i;
            else if (li < l || r < ri) id = n;
            l = Math.min(l, li);
            r = Math.max(r, ri);
        }
        System.out.println(id == n ? -1 : i + 1);
    }
}
